### ISSUE TITLE
Compatibility with Chat Head: Timestamp before Player Head

### ISSUE THREAD

**Issue by Strubbelkopp** [mrbuilder1961/WheresMyChatHistory#62]
**Is your feature request related to a problem? Please describe.**
I feel like it'd make more sense, if the timestamp is displayed before the player head, since also messages without a player head have a timestamp attached to them. This means that the indentation of the timestamps is inconsistent across messages.

**Describe the solution you'd like**
An option for you to display the timestamp in front of the player head

**Describe alternatives you've considered**
I'm not sure whether it'd make more sense to include the option in Chat Patches or Chat Heads. If this is easier doable from the other end, I'll happily open an issue in the Chat Heads repo.

**Comment by mrbuilder1961** [comment]
i think you're right that i would have to do this, but i'm not entirely sure how feasible it would be. i'll look into it

**Comment by mrbuilder1961** [comment]
on second thought, could you attach a photo of what you mean? i'm fairly certain chat heads already let's you indent all messages consistently

**Comment by Strubbelkopp** [comment]
How it currently looks:
![2023-02-05_19 08 31](https://user-images.githubusercontent.com/66599825/216837275-ba73cc65-cba3-4645-9edc-d0314dd0f003.png)
How it looks with the setting from Chat Heads enabled:
![2023-02-05_19 08 55](https://user-images.githubusercontent.com/66599825/216837308-1c454292-11ca-40bf-b846-6c7ef7b5585c.png)
What I'm thinking of:
![2023-02-05_19 08 31-export](https://user-images.githubusercontent.com/66599825/216837316-40267daf-172b-433c-9c5e-11eb2213a10c.png)

With the setting from Chat Heads, the indentation is consistent, however you always have this wide mostly empty column on the left side of the chat, which doesn't look very good in my opinion

**Comment by mrbuilder1961** [comment]
sorry but I don't think that is going to work. chat heads renders the icon from the skin (which makes sense) but for it to work in the best way, it would have to be a sort of text character? and even then there would need to be quite a bit of integration and compatibility work.

now it could possibly be done on chat heads' end: before rendering the icon they would check if the option is enabled and that chat patches is present, then identify the part of the message that is the timestamp (after reconstructing it due to how text is rendered), get the width of that specific timestamp plus the space, and shift the x value by that many pixels to the right. all for an (in my opinion) not very good looking change.

tl;dr: annoying to implement, but if you are really determined you can open an issue on the chat heads repo and link back to this comment.
